packages/odyssey-react-mui/src/labs/DataComponents/useFilterConversion.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { useCallback, useMemo } from "react";
+import { MRT_RowData } from "material-react-table";
+
+import { DataFilter } from "../DataFilters";
+import { DataTableColumn } from "../../DataTable";
+import { UniversalProps, TableProps } from "./componentTypes";
+
+type FilterConversionType = {
+  columns?: TableProps["columns"];
+  filters?: UniversalProps["filters"];
+};
+
+export const useFilterConversion = ({
+  columns,
+  filters,
+}: FilterConversionType) => {
+  const convertFilterSelectOptions = useCallback(
+    (options: DataTableColumn<MRT_RowData>["filterSelectOptions"]) =>
+      options?.map((option) =>
+        typeof option === "string"
+          ? {
+              label: option,
+              value: option,
+            }
+          : {
+              // If the option isn't a string, it must have value and/or option defined
+              // If either is undefined, use the other
+              label: option.label ?? option.value,
+              value: option.value ?? option.label,
+            },
+      ),
+    [],
+  );
```

Comment:
Wish we didn't have to do this. Not pointing out anything other than, it'd be nice to have 2 ways of handling this in the code as it's confusing.